dartsim/src/Base.hh

```diff
@@ -364,6 +376,25 @@ class Base : public Implements3d<FeatureList<Feature>>
     return id;
   }
 
+  public: inline std::size_t AddJointConstraint(DartWeldJointConsPtr _joint)
+  {
+    const std::size_t id = this->GetNextEntity();
+    this->joints.idToObject[id] = std::make_shared<JointInfo>();
+    this->joints.idToObject[id]->constraint = _joint;
+    this->joints.idToObject[id]->type = JointInfo::JointType::CONSTRAINT;
+    // TODO(arjo): Refactor the joints.objectToId to support constraints.
+    // this->joints.objectToID[_joint] = id;
+    // dart::dynamics::SimpleFramePtr jointFrame =
+    //     dart::dynamics::SimpleFrame::createShared(
+    //         _joint->getChildBodyNode(), _joint->getName() + "_frame",
+    //         _joint->getTransformFromChildBodyNode());
+
+    // this->joints.idToObject[id]->frame = jointFrame;
+    // this->frames[id] = this->joints.idToObject[id]->frame.get();
```

Comment:
remove this code or is this relevant to the todo comment?

Reply:
Yeah honestly I'm not sure where `this->frames`  and `this->jointsidToObject` is used. Would be great if someone who knew  the physics subsystem well could advice on whether we need it.